src/main/java/org/apache/sysds/runtime/instructions/fed/BinaryFEDInstruction.java

```diff
@@ -42,11 +47,12 @@ public static BinaryFEDInstruction parseInstruction(String str) {
 		}
 
 		String[] parts = InstructionUtils.getInstructionPartsWithValueType(str);
-		InstructionUtils.checkNumFields(parts, 3, 4);
+		InstructionUtils.checkNumFields(parts, 3, 4, 5);
 		String opcode = parts[0];
 		CPOperand in1 = new CPOperand(parts[1]);
 		CPOperand in2 = new CPOperand(parts[2]);
 		CPOperand out = new CPOperand(parts[3]);
+		boolean federatedOutput = parts.length > 5 && Boolean.parseBoolean(parts[5]);
```

Comment:
does this work? should it not be index 4? or is that the number of threads?

Reply:
Number 4 is the number of threads. I assume that this has been included, otherwise we cannot parse a federated output. I try to only write federated output in the cases where it is true, but I leave room for also writing false to the instruction string. This is the reason why I check if it has been included and parse it if it has been included.